onnxruntime/core/providers/nnapi/nnapi_builtin/builders/op_support_checker.cc

```diff
@@ -25,19 +22,37 @@ struct OpSupportCheckerRegistrations {
   std::unordered_map<std::string, const IOpSupportChecker*> op_support_checker_map;
 };
 
-bool HasExternalInitializer(const InitializedTensorSet& initializers, const Node& node) {
-  for (const auto* node_arg : node.InputDefs()) {
-    const auto& input_name(node_arg->Name());
-    if (!Contains(initializers, input_name))
-      continue;
-
-    const auto& tensor = *initializers.at(input_name);
-    if (tensor.has_data_location() &&
-        tensor.data_location() == ONNX_NAMESPACE::TensorProto_DataLocation_EXTERNAL) {
-      LOGS_DEFAULT(VERBOSE) << "Initializer [" << input_name
-                            << "] with external data location are not currently supported";
+bool HasExternalInitializer(const InitializedTensorSet& initializers, const NodeUnit& node_unit) {
+  const auto is_ext_initializer =
+      [&](const NodeArg& node_arg) {
+        const auto& input_name(node_arg.Name());
+        if (!Contains(initializers, input_name))
+          return false;
+
+        const auto& tensor = *initializers.at(input_name);
+        if (tensor.has_data_location() &&
+            tensor.data_location() == ONNX_NAMESPACE::TensorProto_DataLocation_EXTERNAL) {
+          LOGS_DEFAULT(VERBOSE) << "Initializer [" << input_name
+                                << "] with external data location are not currently supported";
+          return true;
+        }
+
+        return false;
+      };
+
+  const auto& inputs = node_unit.Inputs();
+  for (const auto& input : inputs) {
+    if (is_ext_initializer(input.node_arg))
+      return true;
+
+    if (!input.quant_param)
+      return false;
```

Comment:
>false [](http://example.com/codeflow?start=13&length=5)

why do we have an early return false here? do we not need to consider other inputs in this case?

Reply:
Yes, should be `continue`

Reply:
Is there a unit test that should have caught this error?

Reply:
Will add UT with external initializers for NNAPI and CoreML EP